dms_version/models/dms_version.py

```diff
@@ -0,0 +1,160 @@
+# Copyright 2017-2020 MuK IT GmbH
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import base64
+import zlib
+from collections import defaultdict
+
+import bsdiff4
+
+from odoo import _, api, fields, models, tools
+from odoo.exceptions import ValidationError
+
+
+class DmsVersion(models.Model):
+    _name = "dms.version"
+    _description = "Version"
+    _order = "create_date desc"
+
+    name = fields.Char(string="Name", readonly=True)
+    file_id = fields.Many2one(
+        comodel_name="dms.file", ondelete="cascade", string="File", required=True
+    )
+    storage_id = fields.Many2one(
+        related="file_id.storage_id",
+        comodel_name="dms.storage",
+        string="Storage",
+        readonly=True,
+    )
+    compress_versions = fields.Boolean(
+        related="storage_id.compress_versions", readonly=True
+    )
+    incremental_versions = fields.Boolean(
+        related="storage_id.incremental_versions", readonly=True
+    )
+    previous_version = fields.Many2one(
+        comodel_name="dms.version", ondelete="set null", string="Previous Version"
+    )
+    next_version = fields.Many2one(
+        comodel_name="dms.version", ondelete="set null", string="Next Version"
+    )
+    is_compress = fields.Boolean(string="Compressed", default=False)
+    is_incremental = fields.Boolean(string="Incremental", default=False)
+    size = fields.Integer(string="Size", readonly=True)
```

Comment:
You can get size from attachments when you read the field with `bin_size=True` in context, so you don't need this field.

Reply:
I'm not really sure because in `dms.file` `size` field is necessary too.